### PR TITLE
Ports Tau Ceti's welding animation!

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
A port of [Pariah Station's port](https://github.com/pariahstation/Pariah-Station/pull/665) of [Tau Ceti's welding animation](https://github.com/TauCetiStation/TauCetiClassic/pull/6540)!

Example footage from the Pariah Station port:

image

image

## Why It's Good For The Game
Visual feedback for welding and what is being welded, pretty sprites. 

## Changelog

:cl: UDaV73rus, Tokoriso, dragomagol
add: welding now has an animation!
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
can this be seen from the other side of the wall too?

**Comment by dragomagol** [comment]
I haven't tested that specifically but based on the code I would assume so

**Comment by MTandi** [comment]
A very cool effect, like the fact that it also emits light. 
I think one of the codebases even has it animated for airlocks, so you can see the progress without the progress bar.

File: code/game/objects/items/tools/weldingtool.dm [hunk 1]
```diff
@@ -120,6 +120,11 @@
 	dyn_explosion(src, plasmaAmount/5, explosion_cause = src) // 20 plasma in a standard welder has a 4 power explosion. no breaches, but enough to kill/dismember holder
 	qdel(src)
 
+/obj/item/weldingtool/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
+	target.add_overlay(GLOB.welding_sparks)
```
Comment (Fikou) [inline code/game/objects/items/tools/weldingtool.dm:124]: put it on plasma cutters?

Comment (dragomagol) [reply code/game/objects/items/tools/weldingtool.dm:124]: Shore

**Review by Mothblocks — APPROVED** [review]

**Comment by calliophis-maculiceps** [comment]
would this appear on yourself if you repair a prosthetic limb? if yes then it's really cool

**Review by Twaticus — APPROVED** [review]

**Pull request merged**